### PR TITLE
added testing and pyrdfstore 

### PR DESCRIPTION
**Pull request by cedricdcc**
This PR adds the following:

- Adds Makefile with single testing and graphdb testing capabilities
- yml .github workflow fro PR testing
- submodule folder layout
- the original code and testing from py-rdf-store with some small refactoring on where the variables come from

Issues tackled in this PR:

- #37 
- #43 
- #38 
- #40 
- #41
- #48
- #10

### CODE REVIEW

File: tests/util4tests.py
Comment (marc-portier) [inline tests/util4tests.py:1]: this file is a remnant of our cookiecutter (which could use an update)
considering we have started using `conftest.py` nowadays we might as well include the content of this file inside that one?

**Review by marc-portier — CHANGES_REQUESTED** [review]
think we should do the clean split up prior to merge
as I think other parts of the sema-group are going to use and reference it, and would then lead to possible conflicts 

all other remarks can probably be moved into issues that various people can take up

**Comment by marc-portier** [comment]
also slightly worrying that the automated tests seem to be failing
best to make sure that gets resolved too (strangely my own experience is that tests seem to run nicely locally)

also: the requested split / reposition of the **clean stuff should be accompanied by a reposition of the matching tests I guess

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+TEST_INPUT_FOLDER = Path(__file__).parent / "./input"
+DCT: Namespace = Namespace("http://purl.org/dc/terms/#")
+DCT_ABSTRACT: URIRef = DCT.abstract
+SELECT_ALL_SPO = "SELECT ?s ?p ?o WHERE { ?s ?p ?o . }"
+
+
+enable_test_logging()  # note that this includes loading .env into os.getenv
+
+# TODO: enable this back when pyrdfstore is implemented in sema
```
Comment (marc-portier) [inline tests/conftest.py:19]: mysterious message
what "this" should be enable?

**Comment by marc-portier** [comment]
@cedricdcc 
found some time today to give a hint about that reorganiszation of the cleaning code 
see https://github.com/vliz-be-opsci/py-sema/tree/fix/commons-clean

possibly useful to merge into this branch / PR

File: sema/commons/j2/j2_functions.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    _cache = dict()
+
+    @staticmethod
+    def all():
+        return {
+            "uritexpand": uritexpand,
+            "regexreplace": regexreplace,
+            "map": map_build,
+            # TODO: Check for duplication in filters and the meaning.
+            "xsd": xsd_format,
```
[the remark this answers is not in the document]
Comment (cedricdcc) [reply sema/commons/j2/j2_functions.py:21]: mentioned in #51

File: sema/commons/j2/j2_functions.py
[the remark this answers is not in the document]
Comment (cedricdcc) [reply sema/commons/j2/j2_functions.py:1]: mentioned in #52

File: sema/__init__.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+:synopsis: A library containing functions to work with RDF data in Python.
+
+.. moduleauthor:: "Open Science Team VLIZ vzw" <opsci@vliz.be>
+"""
+
+# TODO: add submodules here
+
+from . import commons
+
+__all__ = ["commons"]
```
[the remark this answers is not in the document]
Comment (cedricdcc) [reply sema/__init__.py:17]: mentioned in #53

**Comment by marc-portier** [comment]
automated tests still failing -- any clue about the why?

**Comment by cedricdcc** [comment]
> automated tests still failing -- any clue about the why?

It has something todo with the coverage tests from @JotaFan.

Installing collected packages: coverage, coverage-badge
Successfully installed coverage-7.5.4 coverage-badge-1.1.1
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.4/x64/bin/coverage-badge", line 5, in <module>
    from coverage_badge.__main__ import main
  File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/site-packages/coverage_badge/__main__.py", line 7, in <module>
    import pkg_resources

**Pull request merged**